lib/travis/build/addons/transifex.rb

```diff
@@ -0,0 +1,134 @@
+require 'travis/build/addons/base'
+
+module Travis
+  module Build
+    class Addons
+      class Transifex < Base
+        CLIENT_VERSION = '==0.10'
+        SUPER_USER_SAFE = true
+        DEFAULTS = {
+          project: 'default',
+          host: 'https://www.transifex.com',
+          debug: false,
+          traceback: false,
+          pull: {
+            all: true,
+            force: true,
+            minimum_perc: 75,
+          }
+        }.freeze
+
+        attr_reader :validation_errors
+
+        def before_script
+          validate! && run_before
+        end
+
+        def after_script
+          run_after if valid?
+        end
+
+        private
+
+          def run_before
+            install
+            configure
+            propose
+          end
+
+          def run_after
+            update
+          end
+
+          def warn_invalid
+            validation_errors.each { |error| sh.echo error, ansi: :red }
+          end
+
+          def install
+            sh.echo 'Installing Transifex Client (beta)', ansi: :yellow
+            sh.cmd "pip install --user transifex-client#{CLIENT_VERSION}", echo: true
+            sh.export 'PATH', '$HOME/.local/bin:$PATH', echo: true
+          end
+
+          def configure
+            sh.cmd "tx init #{tx_init_options}", echo: true
+            sh.cmd "tx set #{tx_set_source_options}", echo: true
+          end
+
+          def propose
+            # TODO: propose translation update
+            sh.cmd "tx #{tx_global_options} pull #{tx_first_pull_options}", echo: true
+            sh.cmd "tx #{tx_global_options} pull #{tx_second_pull_options}", echo: true
+          end
+
+          def update
+            # TODO: update upstream translations
+            sh.cmd "tx #{tx_global_options} push", echo: true
+          end
+
+          def valid?
+            !!@valid
+          end
+
+          def validate!
+            # TODO: define what constitutes a valid configuration, such as relationship to pull requests, disabled
+            # branches, etc.
+            @valid = true
+            @validation_errors = []
+            warn_invalid unless @valid
+            @valid
+          end
+
+          def tx_config
+            @tx_config ||= config[:transifex].deep_symbolize_keys.reverse_merge(DEFAULTS)
+          end
+
+          def tx_init_options
```

Comment:
Ah, nice.  What if we did something similar such that all of the required options were translated directly from the YAML or env vars to `~/.transifexrc`?  a la:

Set all options in YAML:
```
travis setup transifex
```
which results in:
``` yaml
addons:
  transifex:
    username: abcd1234
    password:
      secure: ABCD...ZYXW
```

**OR**

Set username and password in env vars
``` bash
travis env set TRANSIFEX_USERNAME abcd1234
travis env set TRANSIFEX_PASSWORD secretsecret
```
with the complimentary YAML being:
``` yaml
addons:
  transifex: true
```

The idea being that both of these would result in the generation of a `~/.transifexrc` that looks like:

``` ini
[https://www.transifex.com]
hostname = https://www.transifex.com
password = secretsecret
token =
username = abcd1234
```